### PR TITLE
fix(core): implement editor timeout and report error from boundary

### PR DESCRIPTION
**Pull request by joooye34**
1. Implement editor root timeout check
2. Init sentry to report data
3. Use sentry error boundary to report unexpected error from react boundary

image
image

### CODE REVIEW

**Review by Brooooooklyn — APPROVED** [review]

File: packages/frontend/component/src/components/block-suite-editor/index.tsx [hunk 1]
```diff
@@ -77,6 +74,52 @@ const useBlockElementById = (
...
+        expectPageId: page.id,
+        expectGuid: page.spaceDoc.guid,
+        spaceVectors,
+      })
+    );
+  }
+}
+
+const PAGE_ROOT_KEY = Symbol('PAGE_ROOT');
+function loadPageRoot(page: Page) {
```
Comment (EYHN) [inline packages/frontend/component/src/components/block-suite-editor/index.tsx]: this is a react hook function, should be named as `useLoadPageRoot`

File: packages/frontend/component/src/components/block-suite-editor/index.tsx [hunk 2]
```diff
@@ -77,6 +74,52 @@ const useBlockElementById = (
...
+          resolve();
+        });
+        window.setTimeout(() => {
+          disposable.dispose();
+          reject(new NoPageRootError(page));
+        }, 10 * 1000);
+      });
+      Reflect.set(page, PAGE_ROOT_KEY, root$);
+    }
+    use(root$);
```
Comment (EYHN) [inline packages/frontend/component/src/components/block-suite-editor/index.tsx:119]: react hook cannot be used in if

File: packages/frontend/component/src/components/block-suite-editor/index.tsx (hunk 1 above)
Comment (joooye34) [reply packages/frontend/component/src/components/block-suite-editor/index.tsx]: it is not a hook, it is a suspense load function.

File: packages/frontend/component/src/components/block-suite-editor/index.tsx (hunk 2 above)
Comment (joooye34) [reply packages/frontend/component/src/components/block-suite-editor/index.tsx:119]: nope, suspense load function is different with hook function

**Review by Brooooooklyn — APPROVED** [review]

**Pull request merged**